### DISCUSSION TITLE
Tags in Custom Script

### DISCUSSION THREAD

**Discussion by wz4** [netbox-community/netbox#14746] [Q&A] [answered]
On a custom script form, I would like to show tags only related to the Device model. I have been unsuccessful so far finding the right query params. I believe it should be similar to the following, but error is "Object of type ConfigContextModelQuerySet is not JSON serializable." Any advice?

```
    tags = MultiObjectVar(
        label='Tags',
        model=Tag,
        required=False,
        query_params={
            'object_types': Device.pk
        },
    )
```

**Comment by candlerb** [comment, marked as answer]
Untested, but try something like this:

```
        query_params={
            'object_types': [ContentType.objects.get_for_model(Device).pk]
        },
```

or

```
        query_params={
            'object_types': [ContentType.objects.get(app_label='dcim', model='device').pk]
        },
```

I got this from the Netbox [source code](https://github.com/netbox-community/netbox/blob/v3.7.0/netbox/users/tests/test_filtersets.py#L201-L204):


```
    def test_object_types(self):
        object_types = ContentType.objects.filter(model__in=['site', 'rack'])
        params = {'object_types': [object_types[0].pk, object_types[1].pk]}
        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 2)
```

and

```
        object_types = (
            ContentType.objects.get(app_label='dcim', model='site'),
            ContentType.objects.get(app_label='dcim', model='rack'),
            ContentType.objects.get(app_label='dcim', model='device'),
        )
```

**Reply by wz4** [reply]
Thank you, your first suggestion works. I also found a [builtin filter](https://github.com/netbox-community/netbox/blob/develop/netbox/utilities/templatetags/builtins/filters.py#L142) that does effectively the same thing.

That narrows the list by object_type, but eliminates tags that don't have any object_type set. Ideally, I would see both.

**Reply by candlerb** [reply]
That looks like a Jinja2 [template filter](https://docs.netbox.dev/en/stable/plugins/development/templates/#utilities.templatetags.builtins.filters.content_type_id): `{{ foo | content_type_id }}`

If you want a way via the API to filter "tags which are applicable to content type X (including those which are applicable to any content type)" then you can raise it as a feature request.

**Reply by wz4** [reply]
I appreciate the response. I think explicit tag assignments makes more sense for my use case anyway.
